### PR TITLE
Resource for ingredient detail

### PR DESCRIPTION
**Pull request by Dollym89**

### CODE REVIEW

File: data/src/main/java/com/guru/cocktails/data/source/RemoteSource.kt [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 import com.guru.cocktails.domain.model.ListBundle
+import com.guru.cocktails.domain.model.ingredient.IngredientDetail
 import io.reactivex.Single
 
 interface RemoteSource {
     fun getWeatherForCity(city: String): Single<WeatherDto>
 
     fun getNonAlcoList(): Single<ListBundle<IngredientThumb>>
+
+    fun getIngredientDetailForRemote(ingredientId:Int): Single<IngredientDetailDto>
```
Comment (Petrulak) [inline data/src/main/java/com/guru/cocktails/data/source/RemoteSource.kt]: You can rename this to `getIngredientDetail`. We are in the `RemoteSource ` source so we already know it comes from API.

File: data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.guru.cocktails.domain.model.ingredient.IngredientDetail
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class IngredientDetailMapper @Inject constructor(
+        private val ingredientTypeMapper: IngredientTypeMapper
+) : Mapper<IngredientDetail, IngredientDetailDto>() {
+
+    override fun map(from: IngredientDetailDto) = IngredientDetail(
```
Comment (Petrulak) [inline data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt]: we can use `with` and `apply ` to simplify shit.
For example 
```
return with(from){
IngredientDetail(
 id = id ?: invalidInt,
name =name ?: emptyString,
)
}
```

File: data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            id = from.id ?: invalidInt,
+            name = from.name ?: emptyString,
+            nameGrouped = from.nameGrouped ?: emptyString,
+            description = from.description ?: emptyString,
+            imageName = from.imageName ?: emptyString,
+            numShowed = from.numShowed ?: invalidInt,
+            ingredientType = ingredientTypeMapper.map(from.ingredientType?: IngredientTypeDto(invalidInt,emptyString)),
+            voltage = from.voltage ?: invalidInt)
+
+    override fun reverse(to: IngredientDetail) = IngredientDetailDto(
```
Comment (Petrulak) [inline data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt]: same here.

Comment (Dollym89) [reply data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt]: fixed

File: data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt (hunk 2 above)
Comment (Dollym89) [reply data/src/main/java/com/guru/cocktails/data/mapper/IngredientDetailMapper.kt]: Ok, rewritten

File: data/src/main/java/com/guru/cocktails/data/source/RemoteSource.kt (hunk 1 above)
Comment (Dollym89) [reply data/src/main/java/com/guru/cocktails/data/source/RemoteSource.kt]: Renamed

**Pull request merged**